src/util/system.cpp

```diff
@@ -959,6 +959,43 @@ std::string ArgsManager::GetChainName() const
     return CBaseChainParams::MAIN;
 }
 
+static void logArgs(
+    const std::string& prefix,
+    const std::map<std::string, std::vector<std::string>>& args)
+{
+    // don't log passwords and other sensitive values
+    static const std::string blacklist[] = {
```

Comment:
Short of refactoring this PR to use a flag as suggested elsewhere and during the PR Review Club [(log here](https://bitcoin-core-review-club.github.io/16115.html));

- a whitelist would be safer than a blacklist
- there needs to be some mechanism - code docs and perhaps also in the developer notes docs - to help ensure this list is updated on future arg changes